lib/govuk_index/popularity_worker.rb

```diff
@@ -19,12 +19,17 @@ def perform(records, destination_index)
 
     def process_record(record, popularities)
       base_path = record["identifier"]["_id"]
+      title = record["document"]["title"]
       OpenStruct.new(
         identifier: record["identifier"].merge("version_type" => "external_gte", "_type" => "generic-document"),
-        document: record["document"].merge(
+        document: record["document"].merge!(
           "popularity" => popularities.dig(base_path, :popularity_score),
           "popularity_b" => popularities.dig(base_path, :popularity_rank),
           "view_count" => popularities.dig(base_path, :view_count),
+          "autocomplete" => { # Relies on updated popularity. Title is for new records.
+            "input" => title,
+            "weight" => popularities.dig(base_path, :popularity_rank),
```

Comment:
The lower the rank the more popular the document.  The most popular document is ranked 1, the least popular has rank N (where N is the number of documents on GOV.UK).  So by using the rank as the weight, are we sorting suggestions in *reverse* popularity order?  Should this be `-popularities.dig(base_path, :popularity_rank)`?

Reply:
@barrucadu If it should be in reverse, this is not reflected in the results we use server side, see https://www.gov.uk/api/search.json?q=driving&count=10&suggest=autocomplete&fields=popularity_b

It appears if the rank is supposed to be lower the more popular it is, our ranking system has fundamentally failed somewhere. Alternatively this is just poor naming internally and we're getting ranks mixed.

Reply:
Ah, I remember now, we "fix" the ranking order here: https://github.com/alphagov/search-api/blob/3f4d72c5b1fe97a51fb7e12df34b0773a07578cb/lib/indexer/popularity_lookup.rb#L46

I think this was to avoid exactly the same problem when we were trying out using ranks in popularity boosting.

Reply:
I'm currently using the final popularity result post fix. Is this unwise if there are changes coming, or is this fine?

Reply:
I think it's fine